projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -9,12 +9,441 @@
 
 use WP_Error;
 use WP_REST_Request;
+use WP_REST_Response;
 
 /**
  * Class Contact_Form_Endpoint
  * Used as 'rest_controller_class' parameter when 'feedback' post type is registered in \Automattic\Jetpack\Forms\ContactForm\Contact_Form.
  */
 class Contact_Form_Endpoint extends \WP_REST_Posts_Controller {
+	// TODO: add date filter (months) /// `after, before`
+
+	// TODO: add source filter.
+
+	// TODO add bulk actions if needed..
+	// Check Akismet filters there... (contact_form_akismet)
+
+	// TODO: check about `get_responses_permission_check` differences...
+
+	/**
+	 * Registers the REST routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+		parent::register_routes();
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/filters',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_filters' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			)
+		);
+		// TODO: add `args` to the route.
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/bulk_actions',
+			array(
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'bulk_actions' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all distinct sources (posts) and all the distinct available dates that
+	 * any feedback was received, in order to be used as filters in the list.
+	 *
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function get_filters() {
+		// TODO: investigate how we can do this better regarding usage of $wpdb,
+		// performance by querying all the entities, etc..
+		global $wpdb;
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$months = $wpdb->get_results(
+			"SELECT DISTINCT YEAR( post_date ) AS year, MONTH( post_date ) AS month
+			FROM $wpdb->posts
+			WHERE post_type = 'feedback'
+			ORDER BY post_date DESC"
+		);
+		// phpcs:enable
+		$source_ids = Contact_Form_Plugin::get_all_parent_post_ids(
+			array_diff_key( array( 'post_status' => array( 'draft', 'publish', 'spam', 'trash' ) ), array( 'post_parent' => '' ) )
+		);
+		return rest_ensure_response(
+			array(
+				'date'   => array_map(
+					function ( $row ) {
+						return array(
+							'month' => intval( $row->month ),
+							'year'  => intval( $row->year ),
```

Comment:
Nitpick: `(int)` is faster than `intval()`.
```suggestion
							'month' => (int) $row->month,
							'year'  => (int) $row->year,
```